python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -606,40 +614,53 @@ def helptext(f, s):
 
     def _init(self, *args, **kwds):
 
-        # The first thing we do is, if we were called with GPIProxyClassFactory(_existing=some_ganga_object)
-        # to simply create a link between them and us and nothing else.
-        if '_existing' in kwds:
-            # Link the two together in the proxy layer
-            setattr(self, implRef, kwds['_existing'])
-            setattr(kwds['_existing'], proxyObject, self)
-            return
+        ## Zero-th fully initialize self before moving on
+        GPIProxyObject.__init__(self)
 
         ## THE ORDER IN HOW AN OBJECT IS INITIALIZED IS IMPORTANT AND HAS BEEN DOUBLE CHECKED - rcurrie
 
-        ## FIRST INITALIZE A RAW OBJECT INSTANCE CORRESPONDING TO 'pluginclass'
 
-        instance = pluginclass()
+        ## If we're only constructing a raw Proxy to wrap an existing object lets wrap that and return
+        proxy_obj_str = '_proxy_impl_obj_to_wrap'
 
-        ## SECOND WE NEED TO MAKE SURE THAT OBJECT ID IS CORRECT AND THIS DOES THINGS LIKE REGISTER A JOB WITH THE REPO
+        if proxy_obj_str in kwds.keys():
+            instance = kwds[proxy_obj_str]
+            ## Even if we're wrapping something such as here make sure we set all of the proxy related attributes correctly.
+            ## Setting of these attributes shold be done here within this class and should probably be properly be done on proxy construction. aka. here
+        else:
+            ## FIRST INITALIZE A RAW OBJECT INSTANCE CORRESPONDING TO 'pluginclass'
+            ## Object was not passed by construction so need to construct new object for internal use
+            instance = pluginclass()
 
-        # at the object level _impl is a ganga plugin object
+        ## Avoid intercepting any of the setter method associated with the implRef as they could trigger loading from disk
         setattr(self, implRef, instance)
-        setattr(instance, proxyObject, self)
+
+        ## Need to avoid any setter methods for GangaObjects
+        ## Would be very nice to remove this entirely as I'm not sure a GangaObject should worry about it's proxy (if any)
+        instance.__dict__[proxyObject] = self
+        instance.__dict__[proxyClass] = type(name, (GPIProxyObject,), d)
```

Comment:
Only thing left, do we want to be overwriting this? We already set it in the metaobject `__init__` so I'd rather not repeat that here. In future it might make sense to move it here but for now we should just do this in one place.